core/dnsserver/onstartup_test.go

```diff
@@ -0,0 +1,12 @@
+package dnsserver
+
+import "testing"
+
+func TestCheckDomainName(t *testing.T) {
+	if !checkDomainName("an-example.com") {
+		t.Fatalf(`Expected true for "an-example.com"`)
+	}
+	if !checkDomainName("example-?&^%$.com") {
+		t.Fatalf(`Expected false for "example-?&^$.com"`)
+	}
```

Comment:
Why do both test cases have an `!`? 
I think something is wrong here.

Reply:
The second test should fail if `checkDomainName` is true, you have to remove the `!` there. Also your `checkDomainName` function does in fact return true in that second case, which is unintended I think.

Reply:
This is because I'm directly sending a domain-name as an argument to `checkDomainName(domainName)` function and it returns `true` value when domain-name matches Regex and i think we don't need to show  `Expected true for "an-example.com"` output when domain-name matches expected Regex.
whereas, when we pass an wrong `domain-name` as an argument then `checkDomainName(domainName)` returns `false` and I think at that It's needed to print `Expected false for "example-?&^$.com"` that's why I think putting a Not `!` before `checkDomainName(domainName)` gives exprected result.

Reply:
But there is a problem, at the moment your function `checkDomainNames` returns `true` in both cases, which is is unintended and your test is not catching that.

I tried your function in a test program
```
package main

import (
        "fmt"
        "regexp"
)

func checkDomainName(domainName string) bool {
        matchDomain, _ := regexp.Compile("([A-Za-z0-9-]+\\.[A-Za-z0-9-]+)*")
        return matchDomain.MatchString(domainName)
}

func main() {
    output := checkDomainName("an-example.com")
    fmt.Println(output)
    output = checkDomainName("example-?&^%$.com")
    fmt.Println(output)
}
```

And it get the following output 

```
true
true
```

Reply:
Thanks for letting me know,As I can see there is an issue with the Regex so I think I need to improve it.